src/RepositoryChainsArchiver.ts

```diff
@@ -51,105 +63,107 @@ export default class RepositoryChainsArchiver {
+
+              const pathParts = relativePath.split("/");
+              const address = pathParts[2]; // Assuming path format: 'matchType/chainId/addressDirName/...'
+
+              if (!address.startsWith("0x")) {
+                console.log("Skipping non-address folder:", filePath);
+                continue;
+              }
+
+              await this.addFileToPack(currentTarStream, filePath);
+            }
+
+            if (currentTarStream) {
+              currentTarStream.finalize();
+
+              // Upload file
+              const s3Key = `${backupName}/${archiveName}`;
+              await this.uploadFile(localPath, s3Key);
+              uploadedFiles.push({
+                path: s3Key,
+                sizeInBytes: fs.statSync(localPath).size,
+              });
+
+              // Delete file
+              await fs.promises.rm(localPath, {
+                recursive: true,
+                force: true,
+              });
+              console.log(`Deleted local file: ${localPath}`);
+            }
+          }
         }
-
-        const firstTwoChars = address.slice(2, 4);
-        const firstByte = firstTwoChars.toUpperCase();
-
-        // Add the file to the appropriate tar stream
-        await this.addFileToPackStream(matchType, chainId, firstByte, filePath);
-      }
-
-      // After all files have been added, finalize the tar streams
-      for (const archiveName of Object.keys(this.tarStreams)) {
-        this.tarStreams[archiveName].finalize();
       }
+      this.uploadManifest(backupName, uploadedFiles);
+      this.deleteOldBackups();
+      console.log("Deleted old backups");
     } catch (error) {
       console.error(
         `Error processing chains ${this.chainIds.join(",")}:`,
         error
       );
     }
+    console.log("Done");
   }
 
-  private async uploadFile(
-    s3Client: S3Client,
-    bucketName: string,
-    localPath: string,
-    s3Key: string
-  ): Promise<void> {
+  private async uploadFile(localPath: string, s3Key: string): Promise<void> {
     const fileContent = await fs.promises.readFile(localPath);
-    await s3Client.send(
+    await this.s3Client.send(
       new PutObjectCommand({
-        Bucket: bucketName,
+        Bucket: this.s3BucketName,
         Key: s3Key,
         Body: fileContent,
       })
     );
-    console.log(`Uploaded ${s3Key} to ${bucketName}`);
+    console.log(`Uploaded ${s3Key} to ${this.s3BucketName}`);
   }
 
-  private async deleteOldBackups(
-    s3Client: S3Client,
-    bucketName: string
-  ): Promise<void> {
+  private async deleteOldBackups(): Promise<void> {
     // List only the root folders with the prefix
-    const listResponse = await s3Client.send(
+    const listResponse = await this.s3Client.send(
       new ListObjectsV2Command({
-        Bucket: bucketName,
+        Bucket: this.s3BucketName,
         Prefix: "sourcify-repository-",
         Delimiter: "/", // This will only list the root folders
```

Comment:
Doesn't this potentiall delete the new backup? We are calling `deleteOldBackups` after uploading the new one. I couldn't see how we make the distinction between old and new backups in the bucket

Reply:
Look further below. The function always keeps the latest three backups including the new one.

Reply:
👍